packages/dashboard/src/testing/testing-ground/testing-ground.tsx

```diff
@@ -43,6 +160,57 @@ export class TestingGround {
     this.stretchToFit = (e as any).target.checked;
   };
 
+  addAnnotation = (e: Event) => {
+    e.preventDefault();
+    const original = this.dashboardConfiguration;
+    const XANNOTATION: XAnnotation = {
+      color: 'red',
+      value: (e as any).target.value,
+      id: Date.now().toString(),
+    };
+    console.log('Annotation for ', XANNOTATION.value, ' added!');
+    this.dashboardConfiguration = addXAnnotation({
+      dashboardConfiguration: original,
+      widgetId: widgetId,
+      annotation: XANNOTATION,
+    });
+  };
+
+  deleteAnnotation = (e: Event) => {
+    e.preventDefault();
+    const original = this.dashboardConfiguration;
+    const stringInput = (e as any).target[0].value;
+    console.log('Annotation ', stringInput, ' deleted!');
+    this.dashboardConfiguration = deleteXAnnotation({
+      dashboardConfiguration: original,
+      widgetId: widgetId,
+      annotationIdToDelete: stringInput,
+    });
+  };
+
+  editAnnotation = (e: Event) => {
+    e.preventDefault();
+    const original = this.dashboardConfiguration;
+    const dateInput = (e as any).target[1].value;
+    const stringInput = (e as any).target[0].value;
+    console.log('Annotation ', stringInput, ' edited!');
+    const XANNOTATION: XAnnotation = {
+      color: 'red',
+      value: dateInput,
+      id: Date.now().toString(),
+    };
+    this.dashboardConfiguration = editXAnnotation({
+      dashboardConfiguration: original,
+      widgetId: widgetId,
+      oldAnnotationId: stringInput,
```

Comment:
Are we passing an id here to represent the annotation to be edited? Couldn't we just pass the edited annotation which has this id and map in the dashboard action?

Reply:
Refactored!